modules/orchestrator/orchestrator-core/src/main/java/org/apache/airavata/orchestrator/core/validator/impl/GroupResourceProfileValidator.java

```diff
@@ -0,0 +1,155 @@
+import org.apache.airavata.model.experiment.ExperimentModel;
+import org.apache.airavata.model.experiment.UserConfigurationDataModel;
+import org.apache.airavata.model.process.ProcessModel;
+import org.apache.airavata.model.scheduling.ComputationalResourceSchedulingModel;
+import org.apache.airavata.orchestrator.core.validator.JobMetadataValidator;
+import org.apache.airavata.registry.api.RegistryService;
+import org.apache.airavata.registry.api.client.RegistryServiceClientFactory;
+import org.apache.airavata.registry.api.exception.RegistryServiceException;
+import org.apache.thrift.TException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class GroupResourceProfileValidator implements JobMetadataValidator {
+
+    private final static Logger logger = LoggerFactory.getLogger(GroupResourceProfileValidator.class);
+
+    private RegistryService.Client registryClient;
+
+    public GroupResourceProfileValidator() throws TException, ApplicationSettingsException {
+        this.registryClient = getRegistryServiceClient();
+    }
+
+    @Override
+    public ValidationResults validate(ExperimentModel experiment, ProcessModel processModel) {
+        ValidationResults validationResults = new ValidationResults();
+        validationResults.setValidationState(true);
+        try {
+            List<ValidatorResult> validatorResultList = validateGroupResourceProfile(experiment, processModel);
+            for (ValidatorResult result : validatorResultList){
+                if (!result.isResult()){
+                    validationResults.setValidationState(false);
+                    break;
+                }
+            }
+            validationResults.setValidationResultList(validatorResultList);
+        } catch (TException e) {
+            throw new RuntimeException("Error while validating Group Resource Profile", e);
+        }
+        return validationResults;
+    }
+
+    private List<ValidatorResult> validateGroupResourceProfile(ExperimentModel experiment, ProcessModel processModel) throws TException{
+        List<ValidatorResult> validatorResultList = new ArrayList<ValidatorResult>();
+        UserConfigurationDataModel userConfigurationData = experiment.getUserConfigurationData();
+        ComputationalResourceSchedulingModel computationalResourceScheduling = userConfigurationData.getComputationalResourceScheduling();
+
+        String groupResourceProfileId = userConfigurationData.getGroupResourceProfileId();
+        String computeResourceId;
+        if (processModel == null) {
+            computeResourceId = computationalResourceScheduling.getResourceHostId();
+        } else {
+            computeResourceId = processModel.getProcessResourceSchedule().getResourceHostId();
+        }
+
+        List<BatchQueueResourcePolicy> batchQueueResourcePolicies = registryClient.getGroupBatchQueueResourcePolicyList(groupResourceProfileId);
```

Comment:
It's possible for a GroupResourceProfile to have no ComputeResourcePolicy or BatchQueueResourcePolicies. So this logic needs to change slightly so that it only enforces those policies if they exist.